src/core/core.layouts.js

```diff
@@ -3,55 +3,188 @@
 	return helpers.where(array, function(v) {
-		return v.position === position;
+		return v.pos === position;
 	});
 }
 
 function sortByWeight(array, reverse) {
 	array.forEach(function(v, i) {
-		v._tmpIndex_ = i;
+		v.index = i;
+		v.weight = v.box.weight;
 		return v;
 	});
 	array.sort(function(a, b) {
 		var v0 = reverse ? b : a;
 		var v1 = reverse ? a : b;
 		return v0.weight === v1.weight ?
-			v0._tmpIndex_ - v1._tmpIndex_ :
+			v0.index - v1.index :
 			v0.weight - v1.weight;
 	});
-	array.forEach(function(v) {
-		delete v._tmpIndex_;
-	});
 }
 
-function findMaxPadding(boxes) {
-	var top = 0;
-	var left = 0;
-	var bottom = 0;
-	var right = 0;
-	helpers.each(boxes, function(box) {
-		if (box.getPadding) {
-			var boxPadding = box.getPadding();
-			top = Math.max(top, boxPadding.top);
-			left = Math.max(left, boxPadding.left);
-			bottom = Math.max(bottom, boxPadding.bottom);
-			right = Math.max(right, boxPadding.right);
-		}
-	});
+function wrapBoxes(boxes) {
+	var layoutBoxes = [];
+	var i, ilen, box;
+
+	for (i = 0, ilen = (boxes || []).length; i < ilen; ++i) {
+		box = boxes[i];
+		layoutBoxes.push({
+			pos: box.position,
+			box: box,
+			horizontal: box.isHorizontal()
+		});
+	}
+	return layoutBoxes;
+}
+
+function setLayoutDims(layouts, chartArea) {
+	var i, ilen, layout;
+	for (i = 0, ilen = layouts.length; i < ilen; ++i) {
+		layout = layouts[i];
+		// store width used instead of chartArea.w in fitBox
+		layout.width = layout.horizontal
+			? layout.box.fullWidth && chartArea.availableWidth
+			: chartArea.vBoxMaxWidth;
+		// store height used instead of chartArea.h in fitBox
+		layout.height = layout.horizontal && chartArea.hBoxMaxHeight;
+	}
+}
+
+function buildLayoutBoxes(boxes) {
+	var layoutBoxes = wrapBoxes(boxes);
+	var left = filterByPosition(layoutBoxes, 'left');
+	var right = filterByPosition(layoutBoxes, 'right');
+	var top = filterByPosition(layoutBoxes, 'top');
+	var bottom = filterByPosition(layoutBoxes, 'bottom');
+
+	// Sort boxes by weight. A higher weight is further away from the chart area
+	sortByWeight(left, true);
+	sortByWeight(right);
+	sortByWeight(top, true);
+	sortByWeight(bottom);
+
 	return {
-		top: top,
-		left: left,
-		bottom: bottom,
-		right: right
+		lt: left.concat(top),
+		rb: right.concat(bottom),
+		area: filterByPosition(layoutBoxes, 'chartArea'),
+		v: left.concat(right),
+		h: top.concat(bottom)
 	};
 }
 
-function addSizeByPosition(boxes, size) {
-	helpers.each(boxes, function(box) {
-		size[box.position] += box.isHorizontal() ? box.height : box.width;
+function updateMaxPadding(padding, box) {
+	if (box.getPadding) {
+		var boxPadding = box.getPadding();
+		padding.top = Math.max(padding.top, boxPadding.top);
+		padding.left = Math.max(padding.left, boxPadding.left);
+		padding.bottom = Math.max(padding.bottom, boxPadding.bottom);
+		padding.right = Math.max(padding.right, boxPadding.right);
+	}
+}
+
+
+function addSizeByPosition(outerDims, boxes) {
```

Comment:
no need for two blank lines before and after this `function`

Reply:
fixed